command/agent/consul/client.go

```diff
@@ -68,6 +68,11 @@ const (
 
 	// ServiceTagSerf is the tag assigned to Serf services
 	ServiceTagSerf = "serf"
+
+	// deregisterProbationPeriod is the initialization period where
+	// services registered in Consul but not in Nomad don't get registered,
+	// to allow for nomad restoring tasks
+	deregisterProbationPeriod = 10 * time.Minute
```

Comment:
10 minutes seems very long to wait to clear unknown services and checks. Is there any reason this isn't something like 10 seconds? Maybe 1 minute if we're worried about restore times on nodes with lots of allocations?

Reply:
I'll reduce it to 1 minute.  I was being very conservative and wanted to handle when we change service id.  When we change service id, we'd allow the service to be registered under new service id and pass health checks in the grace period before we remove old checks.